src/Symfony/Component/HttpClient/Rfc9111HttpClient.php

```diff
@@ -0,0 +1,616 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\Cache\Adapter\TagAwareAdapterInterface;
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Response\AsyncResponse;
+use Symfony\Component\HttpClient\Response\MockResponse;
+use Symfony\Component\HttpClient\Response\ResponseStream;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+/**
+ * Adds caching on top of an HTTP client (per RFC 9111).
+ *
+ * Known omissions / partially supported features per RFC 9111:
+ *   1. Range requests:
+ *     - All range requests ("partial content") are passed through and never cached.
+ *   2. stale-while-revalidate:
+ *     - There's no actual "background revalidation" for stale responses, they will
+ *       always be revalidated.
+ *   3. min-fresh, max-stale, only-if-cached:
+ *     - Request directives are not parsed; the client ignores them.
+ *
+ * @see https://www.rfc-editor.org/rfc/rfc9111
+ */
+class Rfc9111HttpClient implements HttpClientInterface, ResetInterface
+{
+    use AsyncDecoratorTrait {
+        stream as asyncStream;
+        AsyncDecoratorTrait::withOptions insteadof HttpClientTrait;
+    }
+    use HttpClientTrait;
+
+    private array $defaultOptions = self::OPTIONS_DEFAULTS;
+
+    /**
+     * @param HttpClientInterface      $client                           The decorated client
+     * @param TagAwareAdapterInterface $cache                            The cache to store responses
+     * @param array                    $defaultOptions                   The default options for the client
+     * @param bool                     $sharedCache                      Whether to share the cache between all instances of this client
```

Comment:
what is the actual behavior of this setting ?

If this is about configuring as a shared cache with the meaning of HTTP shared cache (that must not cache private responses), the description of the argument is not clear at all. And otherwise, I have no idea what it could mean at all (as whether the storage is shared between multiple instances of this client depends on whether they use the same cache instance or no)

Reply:
Adressed.